tests/forte/image_annotation_test.py

```diff
@@ -30,21 +31,208 @@ class ImageAnnotationTest(unittest.TestCase):
     """
 
     def setUp(self):
-        self.datapack = DataPack("image")
-        self.line = np.zeros((6, 12))
-        self.line[2, 2] = 1
-        self.line[3, 3] = 1
-        self.line[4, 4] = 1
-        self.datapack.payloads.append(self.line)
-        self.datapack.image_annotations.append(
-            ImageAnnotation(self.datapack, 0)
+        self.datapack_1 = DataPack("image_1")
+        self.line1 = np.zeros((6, 12))
+        self.line1[2, 2] = 1
+        self.line1[3, 3] = 1
+        self.line1[4, 4] = 1
+
+        self.mark = np.zeros((6, 6))
+        self.mark[3, 3:5] = 1
+        self.mark[4, 4:5] = 1
+
+        self.datapack_2 = DataPack("image_2")
+        self.chunk = np.zeros((12, 12))
+        self.chunk[2, :] = 1
+        self.chunk[3, :] = 1
+        self.chunk[4, :] = 1
+
+        self.datapack_1.payloads.append(self.line1)
+        self.datapack_1.payloads.append(self.mark)
+        self.datapack_2.payloads.append(self.chunk)
+
+        # Entries for first Data Pack
+        self.bb1 = BoundingBox(
+            pack=self.datapack_1,
+            height=2,
+            width=2,
+            grid_height=3,
+            grid_width=4,
+            grid_cell_h_idx=1,
+            grid_cell_w_idx=1,
+        )
+        self.datapack_1.add_entry(self.bb1)
+
+        self.bb2 = BoundingBox(
+            pack=self.datapack_1,
+            height=3,
+            width=4,
+            grid_height=5,
+            grid_width=5,
+            grid_cell_h_idx=3,
+            grid_cell_w_idx=3,
+        )
+        self.datapack_1.add_entry(self.bb2)
+
+        # Entries for another image in the same Data Pack
+        self.bb4 = BoundingBox(
+            pack=self.datapack_1,
+            height=3,
+            width=4,
+            grid_height=6,
+            grid_width=6,
+            grid_cell_h_idx=2,
+            grid_cell_w_idx=2,
+            image_payload_idx=1,
```

Comment:
You can add a test to see this `image_payload_idx=1` is actually stored in `self.datapack_1._data_store.get_entry(tid=self.bb4.tid)`. I guess it might still be `0`